### PR TITLE
Feature/validation

### PR DESCRIPTION
**Pull request by MartynasZilinskas**

### CODE REVIEW

File: packages/simplr-forms-core/__tests__/stores/form-store.test.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+    it("registers a field", () => {
+        const formId = "FORM-ID";
+        const fieldId = "FIELD-ID";
+        const initialValue = "INITIAL-VALUE";
+        const formStore = new FormStore(formId);
+
+        formStore.RegisterField(fieldId, initialValue);
+        expect(formStore.GetField(fieldId)).not.toBeUndefined();
+        expect(formStore.HasField(fieldId)).toBe(true);
```
Comment (DovydasNavickas) [inline packages/simplr-forms-core/__tests__/stores/form-store.test.ts]: Check HasField first, then GetField

File: packages/simplr-forms-core/__tests__/stores/form-store.test.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        expect(formStore.GetField(fieldId).InitialValue).toBe(initialValue);
```
Comment (DovydasNavickas) [inline packages/simplr-forms-core/__tests__/stores/form-store.test.ts]: InitialValue is processed before saving it to store, which might change it and fail this test.

File: packages/simplr-forms-core/src/stores/form-store.ts [hunk 3]
```diff
@@ -3,11 +3,12 @@ import { recordify } from "typed-immutable-record";
 import { ActionEmitter } from "action-emitter";
 
 import * as Actions from "./form-store-actions";
-import { FieldState, FieldValue, FieldStateRecord } from "../contracts/field";
+import { FieldState, FieldValue, FieldStateRecord, FormErrorRecord } from "../contracts/field";
```
Comment (DovydasNavickas) [inline packages/simplr-forms-core/src/stores/form-store.ts]: Multiline import

File: packages/simplr-forms-core/__tests__/stores/form-store.test.ts (hunk 2 above)
Comment (MartynasZilinskas) [inline packages/simplr-forms-core/__tests__/stores/form-store.test.ts]: Value is processed in field-base, not in form-store. :)

**Pull request merged**